### ISSUE TITLE
Number of OPEN_CURSORS just keeps growing and growing..

### ISSUE THREAD

**Issue by GasparYYC** [Dani3lSun/apex-plugin-dropzone#14]
### Expected behavior
I would expect that at some point open cursors are closed otherwise the session APEX_PUBLIC_USER keeps holding them open.

### Actual behavior
OPEN_CURSORS keeps increasing and at some point causes the database to issue a ORA-01000  maximum open cursors exceeded

### Steps to reproduce the issue
Just add a lot of files to the dropzone, you don't have to write them to the DB, just to load to the collection.
Even if the collection if flushed via TRUNCATE_COLLECTION or writing the files to the DB the OPEN_CURSORS keep growing. Eventually will get to the limit set by Parameter: OPEN_CURSOR.
Validate how it grows and never shrinks using:
  1  select s.username, max(a.value)
  2  from v$sesstat a, v$statname b, v$session s
  3  where a.statistic# = b.statistic#
  4  and s.sid (+)= a.sid
  5  and b.name = 'opened cursors current'
  6* group by s.username

### APEX version (4.2.6 / 5.0.3)
5.0.3

### Used web server / version and platform (ORDS 3.0.3 / Tomcat 7 / Apache 2.4 / Linux x64)
ORDS 3.0.2.294.08.40, Tomcat 7.0.54, XE 11.2.0.2.0

### Used web browser / version and platform (Chrome 48 Mac / Firefox 44 Windows)
Chrome 55.0.2883.87 m (64-bit)

I suggest to validate:
  In the code close the cursors explicitly and close the results set.

Excellent reference on OPEN_CURSORS on applications leaking cursors... http://www.orafaq.com/node/758

**Comment by Dani3lSun** [comment]
Hey @GasparYYC 
it seems that this is the APEX plugin code itself. I tested it the last hours with several methods and different ORDS versions (up to 3.0.9) but always with an increase of open cursors...
I even tested with an empty AJAX function which do´s nothing
```
FUNCTION ajax_dropzone(p_region IN apex_plugin.t_region,
                                             p_plugin IN apex_plugin.t_plugin)
  RETURN apex_plugin.t_region_ajax_result IS
BEGIN
  RETURN NULL;
END ajax_dropzone;
```
But the result is the same, an increase of open cursors. Normally this functions saves the file in a APEX collection...
So this can´t be the plugin code of dropzone...
But the open cursors are only open until ORDS closes the session from the connection pool (I think max 10-15 mins).
Maybe I have the time to talk to some APEX Dev team member and get a bit more information about it...
Best wishes

**Comment by GasparYYC** [comment]
Hi @Dani3lSun 
 I guess it is possible that after some a period of time a sort of garbage collection kicks in and shuts down the number of open cursors... To be fair I did not wait to see if this was the case.

**BUT** I did find an hourly APEX Scheduler Job:

```
BEGIN
DBMS_SCHEDULER.set_attribute( name => '"ORDS_METADATA"."CLEAN_OLD_ORDS_SESSIONS"', attribute => 'job_action', value => 'ORDS_METADATA.ORDS_INTERNAL.CLEAN_OLD_SESSIONS');

END; 
/
```

So that may be the one, on my production system last night I left if with over 200 OPEN_CURSORS and today is down to 51.


Possibly the workaround is to increase the OPEN_CURSORS parameter to something in the thousands (In my case I chose 3000) cause that should cover the number of interactions I have via DropZone in an hour before the Job kicks in and clears the cursors..

Maybe this is a non-issue and possibly just a recommendation to increase OPEN_CURSORS parameter when using DROPZONE.

Regards
